### PR TITLE
Add dependency on openssl 1.x

### PR DESCRIPTION
**Pull request by ypsah**
libhdfs3 is now compatible with openssl 3.x
This commit is meant to publish an "openssl 1.x compatible" build on
conda-forge.

A later commit will bump the version of the dependency.

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] ~~Reset the build number to `0` (if the version changed)~~
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -31,6 +31,7 @@ requirements:
     - libuuid
     - libxml2
     - krb5
+    - openssl =1
```
Comment (xhochy) [inline recipe/meta.yaml:34]:
```suggestion
    - openssl
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -40,6 +41,7 @@ requirements:
     - libxml2
     - krb5
     - curl
+    - openssl =1
```
Comment (xhochy) [inline recipe/meta.yaml:44]:
```suggestion
```

**Review by xhochy — CHANGES_REQUESTED** [review]
This and a rerender should build it for both openssl versions

File: recipe/meta.yaml (hunk 1 above)
Comment (ypsah) [reply recipe/meta.yaml:34]: I don't get how this works. If we specify `openssl` without a version pin, how is it guaranteed that the version used at build time is the same as the version used at runtime? Even more so if we don't specify `openssl` as a run-time dependency at all.

From what I am reading on [`conda`'s packaging documentation](https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#export-runtime-requirement) we could use `build.run_exports`,

**Comment by martindurant** [comment]
Please note that [`hdfs3`](https://github.com/dask/hdfs3) is marked as archived and has not been updated in years. That project used to be the most common usage of libhdfs3, at least in the python realm. Also, arrow no longer supports libhdfs3 as a filesystem driver. Are people still using this package in other settings? Has libhdfs3 undergone any development since I was involved with it some time ago?

**Pull request closed without merging**